### PR TITLE
refactor(compiler): simplify visitor logic for attributes

### PR DESCRIPTION
**Pull request by atscott**
The logic for computing identifiers, specifically for bound attributes
can be simplified by using the value span of the binding rather than the
source span.

### CODE REVIEW

**Comment by atscott** [comment]
[presubmit](https://test.corp.google.com/ui#cl=332339483)

File: packages/compiler-cli/src/ngtsc/indexer/src/template.ts [hunk 1]
```diff
@@ -191,25 +188,13 @@ class TemplateVisitor extends TmplAstRecursiveVisitor {
...
-    // TODO(ayazhafiz): Remove this when https://github.com/angular/angular/pull/31813 lands.
-    const attributeSrc = attribute.sourceSpan.toString();
-    const attributeAbsolutePosition = attribute.sourceSpan.start.offset;
-
-    // Skip the bytes of the attribute name so that there are no collisions between the attribute
-    // name and expression identifier names later.
-    const nameSkipOffet = attributeSrc.indexOf(attribute.name) + attribute.name.length;
-    const expressionSrc = attributeSrc.substring(nameSkipOffet);
-    const expressionAbsolutePosition = attributeAbsolutePosition + nameSkipOffet;
+    if (attribute.valueSpan === undefined) {
```
Comment (AndrewKushnir) [inline packages/compiler-cli/src/ngtsc/indexer/src/template.ts:191]: nit: I think it'd be great to add a quick comment on why `valueSpan` absence causes a noop.

Comment (AndrewKushnir) [reply packages/compiler-cli/src/ngtsc/indexer/src/template.ts:191]: Quick question: is it a safety measure or this is something that we can repro and have a test for?

**Review by AndrewKushnir — APPROVED** [review]
LGTM (just a couple comments) 👍

File: packages/compiler-cli/src/ngtsc/indexer/src/template.ts (hunk 1 above)
Comment (atscott) [reply packages/compiler-cli/src/ngtsc/indexer/src/template.ts:192]: Added comment and test as well as verified that the test passed on `master` (i.e., that the behavior is the same with and without this change)

Comment (AndrewKushnir) [reply packages/compiler-cli/src/ngtsc/indexer/src/template.ts:192]: Thanks @atscott 👍

**Review by ayazhafiz — APPROVED** [review]

**Review by kyliau — APPROVED** [review]

**Comment by kyliau** [comment]
presubmit: 	http://test/OCL:332582371:BASE:332582379:1600492659981:ae5051a3

**Pull request closed without merging**